src/app/pages/TransactionsPage/index.tsx

```diff
@@ -3,19 +3,44 @@ import { useTranslation } from 'react-i18next'
 import Card from '@mui/material/Card'
 import CardHeader from '@mui/material/CardHeader'
 import CardContent from '@mui/material/CardContent'
+import Divider from '@mui/material/Divider'
 import { PageLayout } from '../../components/PageLayout'
 import { Transactions } from '../../components/Transactions'
 import { useGetEmeraldTransactions } from '../../../oasis-indexer/api'
-import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
+import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE, REFETCH_INTERVAL } from '../../config'
 
 const limit = NUMBER_OF_ITEMS_ON_SEPARATE_PAGE
 
 export const TransactionsPage: FC = () => {
   const { t } = useTranslation()
-  const transactionsQuery = useGetEmeraldTransactions({ limit: limit })
+  const transactionsQuery = useGetEmeraldTransactions(
+    { limit: limit },
+    {
+      query: {
+        refetchInterval: REFETCH_INTERVAL,
+        structuralSharing: (previousState, nextState) => {
+          if (!previousState) {
+            return nextState
+          }
+          return {
+            ...nextState,
+            data: {
+              transactions: nextState.data?.transactions?.map(tx => {
+                return {
+                  ...tx,
+                  markAsNew: !previousState.data?.transactions?.some(prevTx => prevTx.hash === tx.hash),
+                }
+              }),
```

Comment:
Looks to me like spread should be in deeper level. Wouldn't this discard all previous transactions in `previousState`?

Reply:
In what way it can discard previous transactions? you mean like I am mutating it somewhere or not returning prev state ?

Reply:
spread seems Ok
![Screenshot 2023-01-03 at 12 48 49](https://user-images.githubusercontent.com/891392/210351715-44decdf6-d72e-4952-b369-66edc05c34c5.png)

Reply:
By not returning prevState. But I guess that is expected behaviour.